Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp

```diff
@@ -277,10 +277,22 @@ void ThreadedCompositor::renderLayerTree()
     m_scene->applyStateChanges(states);
     m_scene->paintToCurrentGLContext(viewportTransform, FloatRect { FloatPoint { }, viewportSize }, m_flipY);
 
-    m_context->swapBuffers();
+    WebCore::Damage boundsDamage;
+    const auto& frameDamage = ([this, &boundsDamage]() -> const WebCore::Damage& {
+        const auto& damage = m_scene->lastDamage();
+        if (m_client.settings().propagateDamagingInformation() && !damage.isInvalid()) {
+            if (m_client.settings().unifyDamagedRegions()) {
```

Comment:
Same for this setting, maybe we can unify both in a DamageFlags OptionSet that the threaded compositor receives at construction. I don't think those settings are expected to change at runtime, right?

Reply:
They may change at runtime. For example in the MiniBrowser from the settings page (or in Ephy when I manage to update and land the patch to add the feature flags to the settings dialog). Being able to toggle them has been useful at times during development.